### ISSUE TITLE
id attribute not used by Form->radio

### ISSUE THREAD

**Issue by scarabedore** [cakephp/cakephp#16835]
### Description

# How to reproduce
Using the Form helper, create a radio control and specify an `id` attribute. In the generated HTML, `id`s of the radio element are built using the `$fieldName` argument instead of the `id` attribute.

```php
<?= $this->Form->radio(
    'is_company_employer',
    ['yes', 'no'],
    ['id' => 'inputIsEmployer']
) ?>
```

produces
```html
<input type="hidden" name="is_company_employer" id="inputIsEmployer" value="">
<label for="is-company-employer-0">
    <input type="radio" name="is_company_employer" value="0" id="is-company-employer-0">
    yes
</label>
<label for="is-company-employer-1">
    <input type="radio" name="is_company_employer" value="1" id="is-company-employer-1">
    no
</label>
```

# Expected behavior
Ids to be built using the `id` argument.
```html
<input type="hidden" name="is_company_employer" id="inputIsEmployer" value="">
<label for="inputIsEmployer-0">
    <input type="radio" name="is_company_employer" value="0" id="inputIsEmployer-0">
    yes
</label>
<label for="inputIsEmployer-1">
    <input type="radio" name="is_company_employer" value="1" id="inputIsEmployer-1">
    no
</label>
```

### CakePHP Version

4.4.6

### PHP Version

8.0.19

**Comment by scarabedore** [comment]
With Cake PHP 4.3.10, the `id` attribute is honored in the generated HTML. Thus I don't know whether it's a bug, or a change in the API.

**Comment by scarabedore** [comment]
Issue seems to be coming from `FormHelper.php`, line 1574:
```php
if (!isset($attributes['type']) && isset($attributes['name'])) {
    unset($attributes['id']);
}

$radio = $this->widget('radio', $attributes);
```

Namely, `$attributes['type']` is not set to `'radio'`, thus the `id` is removed.
A workaround is to explicitely set the `type` attribute when creating the control:
```php
<?= $this->Form->radio(
    'is_company_employer',
    ['yes', 'no'],
    [
        'id' => 'inputIsEmployer',
        'type' => 'radio',
    ]
) ?>
```

**Comment by markstory** [comment]
The lines you mentioned were added as part of a fix for #16461 in #16475. It looks like we were missing a test case for this situation.
